### PR TITLE
5 - Make more use of `let_chains`

### PR DESCRIPTION
**Pull request by c410-f3r**
Continuation of #94376.

cc #53667

### CODE REVIEW

File: compiler/rustc_passes/src/check_attr.rs [hunk 1]
```diff
@@ -1113,22 +1113,20 @@ impl CheckAttrVisitor<'_> {
...
+            self.tcx.struct_span_lint_hir(UNUSED_ATTRIBUTES, hir_id, attr.span, |lint| {
+                lint.build(
+                    "`must_use` attribute on `async` functions \
+                            applies to the anonymous `Future` returned by the \
+                            function, not the value within",
+                )
+                .span_label(
+                    span,
+                    "this attribute does nothing, the `Future`s \
+                            returned by async functions are already `must_use`",
```
Comment (estebank) [inline compiler/rustc_passes/src/check_attr.rs:1126]:
```suggestion
                    "`must_use` attribute on `async` functions \
                     applies to the anonymous `Future` returned by the \
                     function, not the value within",
                )
                .span_label(
                    span,
                    "this attribute does nothing, the `Future`s \
                     returned by async functions are already `must_use`",
```

File: compiler/rustc_passes/src/dead.rs [hunk 2]
```diff
@@ -683,34 +683,33 @@ impl<'tcx> DeadVisitor<'tcx> {
...
+                        .iter()
+                        .map(|(trait_id, _)| format!("`{}`", self.tcx.item_name(*trait_id)))
+                        .collect::<Vec<_>>()
+                        .join(" and ");
+                    let plural_s = pluralize!(ign_traits.len());
+                    let article = if ign_traits.len() > 1 { "" } else { "a " };
+                    let is_are = if ign_traits.len() > 1 { "these are" } else { "this is" };
+                    let msg = format!(
+                        "`{}` has {}derived impl{} for the trait{} {}, but {} \
+                            intentionally ignored during dead code analysis",
```
Comment (estebank) [inline compiler/rustc_passes/src/dead.rs:700]:
```suggestion
                        "`{}` has {}derived impl{} for the trait{} {}, but {} \
                         intentionally ignored during dead code analysis",
```

**Review by estebank — APPROVED** [review]
Would you have time to also fix the indentation of the highlighted string literals?
Either way, r=me.

**Comment by c410-f3r** [comment]
> Would you have time to also fix the indentation of the highlighted string literals? Either way, r=me.

Done

**Pull request merged**